django/contrib/contenttypes/fields.py

```diff
@@ -160,20 +160,33 @@ def _check_content_type_field(self):
     def get_cache_name(self):
         return self.name
 
-    def get_content_type(self, obj=None, id=None, using=None):
+    def get_content_type(self, obj=None, id=None, using=None, model=None):
         if obj is not None:
             return ContentType.objects.db_manager(obj._state.db).get_for_model(
                 obj, for_concrete_model=self.for_concrete_model
             )
         elif id is not None:
             return ContentType.objects.db_manager(using).get_for_id(id)
+        elif model is not None:
+            return ContentType.objects.get_for_model(
+                model, for_concrete_model=self.for_concrete_model
+            )
         else:
             # This should never happen. I love comments like this, don't you?
             raise Exception("Impossible arguments to GFK.get_content_type!")
 
     def get_prefetch_queryset(self, instances, queryset=None):
-        if queryset is not None:
-            raise ValueError("Custom queryset can't be used for this lookup.")
+        if queryset is not None and not isinstance(queryset, dict):
+            raise ValueError("Custom queryset must be a dict for this lookup.")
+
+        if queryset is None:
+            queryset = {}
+
+        custom_queryset_dict = {}
+
+        for model_cls, ct_queryset in queryset.items():
+            ct_id = self.get_content_type(model=model_cls).pk
```

Comment:
We should pass `using` from the queryset
```suggestion
            ct_id = self.get_content_type(model=model_cls, using=ct_queryset.db).pk
```